### PR TITLE
[TT-1606] Separated analytics storage

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Added configuration options `enable_separate_analytics_store` and `analytics_store` to have a separated analytics storage.
Codewise, it works in the same way as `cache_storage` and `enable_separate_cache`.

**DOCS WIP**

## Related Issue

https://tyktech.atlassian.net/browse/TT-1606

## Motivation and Context

If Redis runs out of memory during a high workload (where analytics would probably have a big part on it), it will make the entire Tyk cluster unable to continue functioning.
To minimize the impact in those situations, the idea is to add an option to have a separated Redis / Redis cluster for analytics.

## How This Has Been Tested

- Checked if the whole analytics flow (gw - request - pump receive the record) is still working without any modification.
- Checked if we set `enable_separate_analytics_store:true` and configure `analytics_store` to another Redis, everything should work on the gateway side but the pump shouldn't process anything. If we change the pump config to this new Redis, it should purge all the records.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [x] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: storage/redis_cluster.go [hunk 1]
```diff
@@ -80,44 +82,55 @@ func WaitConnect(ctx context.Context) bool {
 	}
 }
 
-func singleton(cache bool) redis.UniversalClient {
+func singleton(cache, analytics bool) redis.UniversalClient {
```
Comment (buger) [inline storage/redis_cluster.go:85]: Are you sure we need boolean flag here?
How separate cache storage works then? Does it mean it currently broken?

Comment (tbuchaillot) [reply storage/redis_cluster.go:85]: @buger what do you mean? the cache flag is also passed to singleton func.

Comment (buger) [reply storage/redis_cluster.go:85]: Damn not sure how I read that. Indeed.

**Comment by buger** [comment]
Tests are failing

**Comment by tbuchaillot** [comment]
> Tests are failing

solved @buger

**Review by sredxny — APPROVED** [review]
LGTM

**Review by matiasinsaurralde — APPROVED** [review]
Looks good

**Pull request merged**